### DISCUSSION TITLE
I have a Qustion . How I Can See the Pseudo-Class of Control that not Customer?

### DISCUSSION THREAD

**Discussion by hubing53** [AvaloniaUI/Avalonia#8980] [General]
The Docu,emt  I Can't see?

**Comment by timunie** [comment]
Can you give us a bit more context? Where do you want to see it? In DevTools? In code behind? In XAML? 

Are you the owner of this control?

**Reply by hubing53** [reply]
Yes I Can .In Code behind. Can you tell me where i look at SomeWhere about the other control has been Pseudo-class that defined by the control itself .like documents or like  DevTools etc.
![image](https://user-images.githubusercontent.com/112156001/190847258-2eafa2ec-0a9f-4a1e-b149-d98dcf10e64c.png)

**Reply by timunie** [reply]
## Reading the currently applied pseudo classes

Dev-Tools: press F12 in debug mode

In code behind: use PseudoClasses property to listen to applied pseudo classes (http://reference.avaloniaui.net/api/Avalonia/StyledElement/20D4B58E)

## how to know all possible pseudo classes

I am used to look up the original styles to see which pseudo classes they apply and how to style them properly. Example for ComboBox: https://github.com/AvaloniaUI/Avalonia/blob/stable/0.10.x/src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml#L168-L236

## Improving the docs?

I somehow feel we need to improve the docs here. So for each control we might want to add a section about which pseudo classes are there. But this will take some time. I'll ask the core team about it.

**Comment by jayars** [comment]
I found a similar question ([6178](https://github.com/AvaloniaUI/Avalonia/discussions/6178)) from a year ago. 

Was wondering if the answer is still the same or if things have changed since then.

For me, this is how I'm currently navigating my way around the library of Avalonia controls:

1. Find the control's API page (ie, [Checkbox](http://reference.avaloniaui.net/api/Avalonia.Controls/CheckBox/))
2. Inspect the **Syntax** and check for `PseudoClasses` attributes)
    1. ie, nothing
4. Repeat recursively for the entire inheritance chain. ie
    1. [ToggleButton](http://reference.avaloniaui.net/api/Avalonia.Controls.Primitives/ToggleButton/)
        1. `checked`, `unchecked`, `indeterminate`
    2. [Button](http://reference.avaloniaui.net/api/Avalonia.Controls/Button/)
        1. `pressed`
    3. [ContentControl](http://reference.avaloniaui.net/api/Avalonia.Controls/ContentControl/)
    4. [TemplatedControl](http://reference.avaloniaui.net/api/Avalonia.Controls.Primitives/TemplatedControl/)
    5. [Control](http://reference.avaloniaui.net/api/Avalonia.Controls/Control/)
    6. [InputElement](http://reference.avaloniaui.net/api/Avalonia.Input/InputElement/)
        1. `disabled`, `focus`, `focus-visible`, `pointerover`
    7. [Interactive](http://reference.avaloniaui.net/api/Avalonia.Interactivity/Interactive/) (**Can probably stop when we reach here?**)
    8. [Layoutable](http://reference.avaloniaui.net/api/Avalonia.Layout/Layoutable/)
    9. [Visual](http://reference.avaloniaui.net/api/Avalonia/Visual/)
    10. [StyledElement](http://reference.avaloniaui.net/api/Avalonia/StyledElement/)
    11. [Animatable](http://reference.avaloniaui.net/api/Avalonia.Animation/Animatable/)
    12. [AvaloniaObject](http://reference.avaloniaui.net/api/Avalonia/AvaloniaObject/)

aka, `CheckBox` has the following pseudo classes: `checked`, `unchecked`, `indeterminate`, `pressed`, `disabled`, `focus`, `focus-visible`, `pointerover`

Not sure if there an easier way (perhaps in the docs or in the Rider/VS extension/plugin that I missed) to identify all valid pseudo classes for a control?

**Reply by timunie** [reply]
I found the same as you found. Tbh for now I always just looked up the original styles to find to pseudo classes. I don't know if we can aggregate all of them in the API easily. That would be a great help I think.
